conans/client/client_cache.py

```diff
@@ -184,7 +184,7 @@ def conan_manifests(self, conan_reference):
         digest_path = self.digestfile_conanfile(conan_reference)
         if not os.path.exists(digest_path):
             return None, None
-        export_sources_path = self.export_sources(conan_reference)
+        export_sources_path = self.export_sources(conan_reference, short_paths=None)
```

Comment:
``short_paths=None`` means use short paths if the link already exists. It is not that it is not needed it is that the folder should exist, and the link should be there if short_paths